src/Symfony/Bundle/TwigBundle/CacheWarmer/TemplateCacheWarmer.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\CacheWarmer;
+
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+
+/**
+ * Generates the Twig cache for all templates.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class TemplateCacheWarmer implements CacheWarmerInterface
+{
+    private $twig;
+    private $iterator;
+
+    public function __construct(\Twig_Environment $twig, \Iterator $iterator)
```

Comment:
true, it's currently broken anyway: it passed an InteratorAggregate but expect an Iterator